lib/src/backend/invoker.dart

```diff
@@ -348,6 +352,12 @@ class Invoker {
         await _outstandingCallbacks.noOutstandingCallbacks;
         if (_timeoutTimer != null) _timeoutTimer.cancel();
 
+        if (liveTest.state.result != Result.success &&
+            _runCount < liveTest.test.metadata.retry + 1) {
+          _onRun();
```

Comment:
have you explored what happens to the stack trace for these tests? I'm curious how things look when we go through the same Chain.capture() multiple tmes